scripts/core/editor3/reducers/toolbar.jsx

```diff
@@ -223,6 +229,38 @@ const applyEmbed = (state, code) => {
     return onChange(state, editorState);
 };
 
+
+/**
+ * @ngdoc method
+ * @name toggleParagraphMark
+ * @param {Object} state
+ * @return {Object} returns new state
+ * @description Enable/Disable the paragraph marks
+ */
+const toggleParagraphMark = (state) => {
+    const {paragraphMark, editorState} = state;
+    const spellcheck = ng.get('spellcheck');
+    const decorators = [LinkDecorator];
+    const newParagraphMark = !paragraphMark;
+
+    if (state.spellcheckerEnabled || spellcheck.isAutoSpellchecker) {
```

Comment:
Why do you need to know about the spellchecker here? Can't you retrieve all existing decorators, add yours and create a composite without explicitly listing the other ones?